docs/core/project-sdk/msbuild-props.md

```diff
@@ -148,6 +149,18 @@ The `CopyLocalLockFileAssemblies` property is useful for plugin projects that ha
 > [!TIP]
 > Alternatively, you can use `dotnet publish` to publish the class library. For more information, see [dotnet publish](../tools/dotnet-publish.md).
 
+### PreserveCompilationContext
+
+The `PreserveCompilationContext` property allows the application to compile more code at run time using the same settings that were used at build time. The assemblies referenced at build time will be copied into the *ref* subdirectory of the output directory. The names of the reference assemblies are stored in the application's `.deps.json` file along with the options passed to the compiler. You can retrieve this information using the <xref:Microsoft.Extensions.DependencyModel.DependencyContext.CompileLibraries?displayProperty=nameWithType> and <xref:Microsoft.Extensions.DependencyModel.DependencyContext.CompilationOptions?displayProperty=nameWithType> properties.
+
+This functionality is mostly used internally by ASP.NET MVC and Razor pages.
```

Comment:
```suggestion
This functionality is mostly used internally by ASP.NET Core MVC and Razor pages to support runtime compilation of Razor files.
```

Reply:
If you're feeling up to it, there's also a `PreserveCompilationReferences` option which is a more fine grained option that controls if the refs folder is copied to the output, but keeps the deps file.